Update readme.md (#22)
w3f/polkadot-deployer: README.md

# polkadot-deployer

polkadot-deployer is a general tool for deploying Polkadot nodes, aiming to make it easy to deploy a network of nodes. To learn more about Polkadot explore [the wiki](https://wiki.polkadot.network)
or [join the conversation at Riot](https://riot.im/app/#/room/#polkadot-watercooler:matrix.org).

> polkadot-deployer allows you to create local or remote cloud deployments of polkadot. Currently it supports local deployments using Kind and remote deployments using Google Cloud Platform for the infrastructure and Cloudflare for the DNS settings that make your network accessible through websockets RPC.

Author: Will separate to new PR

## Requirements

The tool is meant to work on Linux and MacOS machines. In order to be able to use the tool you will require to have installed recent versions of [node](https://nodejs.org/en/download/) (developed and tested with `v10.7.0` and `v10.15.1) and
 [docker](https://docs.docker.com/install/) for local deployments (developed and tested with `18.09.5`). Once installed, you should also be able to [run `docker` as a regular user](https://docs.docker.com/install/linux/linux-postinstall/#manage-docker-as-a-non-root-user). See the [Troubleshooting section](#troubleshooting) in case you have problems running the tool.

## Local deployments (TL;DR method)

In order to deploy a number of polkadot nodes locally, we recommend using [kubernetes-sigs/kind](https://github.com/kubernetes-sigs/kind). The required steps to successfully deploy polkadot validator locally are as follows:

1. Download the latest polkadot deployer from git issuing the following command and change directory to polkadot-deployer:  

> 	```git clone git@github.com:w3f/polkadot-deployer.git```

Author: I had removed the npm package because at the time it was broken. Also, should we refer npm or yarn?

> 	```git clone git@github.com:w3f/polkadot-deployer.git```

Reviewer: I think yarn would be better

> 2. You can deploy the polkadot deployer either using the interactive menu or by using a config file. 

Author: I didn't know that. But I think its a better entry point to run the deployer with as little automation as possible.

  * In order to create through the interactive menu issue the following command:  

> 	```sudo node . create --verbose```

Author: removed.

>   * In order to deploy polkadot using a the preset configuration file: my-local-testnet.json and issue the following command:  

Reviewer: The filename needs to be updated here too:
```suggestion
  * In order to deploy polkadot using the preset configuration file: `config/create.local.sample.json` and issue the following command:  
```

	```sudo node . create --config my-local-testnet.json --verbose```
	The process will start creating an instance of polkadot inside a your local kubernetes cluster that will be created as part of the procedure using [kubernetes-sigs/kind](https://github.com/kubernetes-sigs/kind). The entire procedure will take some time, so it might be a good idea to get some coffee at this point.

> 3. Once the local cluster is created, a kubeconfig file will be created in /root/.kube/ with a prefix of kind-config followed by the name of the created cluster (eg: /root/.kube/kind-config-my-demo-testnet). This file should be moved or copied to your local .kube directory. You can archive this by issuing the following command:    ```sudo cp /root/.kube/kind-config-my-local-testnet ~/.kube/kind-config-my-local-testnet```

Author: removed.

> 4. Attach kubectl to this kubeconfig file by issuing the command: ```export KUBECONFIG="~/.kube/kind-config-my-local-testnet"```  You can verify your installation by using the following commands:  ```kubectl cluster-info```  to see an overview of your local installation and ```kubectl get pods``` to see the pods running on your local cluster. You can also view all your local deployments using the command: ```sudo node . list```

Author: updated. I used this method because for intro tutorials etc I think it's better to use as explicit as possible with as little "magic" as possible.

5. At this point you can attach to the local polkadot web socket by visiting the websockets endpoint available at ws://127.0.0.1:11000 Furthermore you at this point you will be presented with the raw seeds for the created accounts, including the nodeKey, peerId, stash address and seed etc.

6. Once you are done with your local deployment of polkadot, you can delete your deployment using the destroy [name] command: ```node . destroy testnet5```
More information on the polkadot-deployer usage commands can be found in the [usage](#usage) section.

Check the [Troubleshooting section](#troubleshooting) if something goes wrong with the installation.

## Remote deployments

To perform a remote deployment of polkador to a public cloud provider we will follow the same general path. We will need to specify a number of extra attributes inside the json file with  extra information such as our credentials etc. In order to be able to deploy remotely you will need:

> * A Linux machine to run this tool (macOS may fail).

Reviewer: I would add a link here to the troubleshooting section below.
```suggestion
* A Linux machine to run this tool (macOS may fail, see the [Troubleshooting section](#troubleshooting) in case you have problems running the tool).
```
